runtime/extism.h

```diff
@@ -3,6 +3,10 @@
 #include <stdint.h>
 #include <stdbool.h>
 
+#ifdef __cplusplus
```

Comment:
This file is generated by cbindgen, I think you can do this by calling [Builder::with_cpp_compat](https://docs.rs/cbindgen/latest/cbindgen/struct.Builder.html#method.with_cpp_compat) in `build.rs`: https://github.com/extism/extism/blob/main/runtime/build.rs